pyActigraphy/io/tools.py

```diff
@@ -0,0 +1,193 @@
+        """
+        initialize the reader class.
+        Each instance of calss coressponds to a block of
+        binary data.
+
+        Parameter
+        ---------
+        data: bytes
+            data to interprete
+        endianess: string
+        """
+        if not isinstance(data, bytes):
+            raise TypeError("data must be bytes")
+
+        self.__data = data
+        self.__offset = 0
+        self.endianess = endianess
+
+    @property
+    def offset(self):
+        return self.__offset
+
+    @offset.setter
+    def offset(self, offset):
+        if not isinstance(offset, int):
+            raise TypeError("offset must be int")
+        if offset < 0:
+            offset = self.size + offset
+        if offset >= self.size or offset < 0:
+            raise IndexError("offset out of range")
+        self.__offset = offset
+
+    @property
+    def endianess(self):
+        return self.__endianess
+
+    @endianess.setter
+    def endianess(self, endianess):
+        if not isinstance(endianess, str):
+            raise TypeError("endianess must be str")
+        if len(endianess) != 1 or endianess not in "@=<>!":
+            raise ValueError("endianess must be one of struct character")
+        self.__endianess = endianess
+
+    @property
+    def size(self):
+        return len(self.__data)
+
+    def __rshift__(self, increment):
+        """
+        increases offset by increment
+        """
+        if not isinstance(increment, int):
+            raise TypeError("increment must be int")
+        if self.__offset + increment > len(self.__data):
+            raise IndexError("offset out of range")
+        self.__offset += increment
+
+    def __lshift__(self, increment):
+        """
+        decrease offset by increment
+        """
+        if not isinstance(increment, int):
+            raise TypeError("increment must be int")
+        if self.__offset < increment:
+            raise IndexError("offset out of range")
+        self.__offset -= increment
+
+    def __iconcat__(self, data):
+        """
+        appends new byte array to data
+        """
+        if not isinstance(data, bytes):
+            raise TypeError("data must be bytes")
+        self.__data += data
+
+    def unpack(self, type, size=1, offset=None):
+        """
+        unpack and returns corresponding data, increasing
+        the offset
+
+        Parameters
+        ----------
+        type: str
+            type of data to unpack
+            if type is char, standard struct types used
+            if type is string, type is interpreted
+            if type is empty, raw data is returned
+        size: int
+            number of data to interpret
+            if equal to 1, returns one instance
+            if >1 returns list of instances
+        offset: int, optional
+            if precised, data readed at given offset, without
+            advancing internal one
+        """
+        if not isinstance(type, str):
```

Comment:
@nbeliy : for the sake of code performance, do you think it is necessary to perform these checks every time you call the `unpack` function (which is a lot of times!)? I agree about type checking (although it is maybe very python-oriented) but only when instantiating a class object for example.

Reply:
I doubt it influence the performance significally. to test